src/tcgwars/logic/impl/gen3/DeltaSpecies.groovy

```diff
@@ -1170,17 +1170,8 @@ public enum DeltaSpecies implements LogicCardInfo {
           attackRequirement {}
           onAttack {
             damage 20
-            if (self.cards.filterByType(ENERGY).findAll {
-              it.asEnergyCard().containsTypePlain(W)
-            }) {
-              damage 20
-            }
-
-            if (self.cards.filterByType(ENERGY).findAll {
-              it.asEnergyCard().containsTypePlain(P)
-            }) {
-              discardDefendingSpecialEnergy(delegate)
-            }
+            if (self.cards.filterByEnergyType(W)) { damage 20 }
```

Comment:
How do both the old version and this new version handle Energy that provide multiple types, but only one type at a time?

Reply:
Should take a single rainbow as sufficient for both effects.

> Q. When using Houndoom's "Black Fire" attack, what happens if I discard an Energy card that provides multiple types of Energy at the same time?
>
> A. If the Energy card you discarded provides all types (for example, Rainbow Energy), then both effects would trigger - the "Black Fire" attack would to 70 damage AND apply the Burned Special Condition to the Defending Pokemon. (DP:Great Encounters FAQ; Feb 21, 2008 PUI Rules Team)

> Q. If you have a Rainbow Energy attached to a Pokemon using one of the Miracle Sphere trainers, can you do both conditions of the card (i.e. both Confused *AND* removing damage for Alpha)? The Miracle Sphere cards specify "if you have xxx Basic Energy cards..." So Rainbow will not work for the extra requirements of the Miracle Spheres, right?
> 
> A. Right. (May 8, 2003 WotC Chat, Q1523 & Q1542)

Couldn't find a specific ruling from EX Era, but those two follow line with each check happening independently, so a single rainbow and all other involved energies being colorless should still trigger all effects.

Reply:
So "1 at a time" only refers to attack energy cost it seems?

Reply:
Yup, it's geared towards the base energy cost printed in the cards. No Rainbow+DCE shenanigans :P